token-metadata-creator/src/Cardano/Metadata/Types.hs

```diff
@@ -372,6 +388,13 @@ validateMetadataDescription :: MonadFail f => Text -> f Description
 validateMetadataDescription = fmap Description .
     validateMaxLength 500
 
+validateMetadataDecimals :: MonadFail f => Scientific -> f Decimals
+validateMetadataDecimals s =
+  case toBoundedInteger s of
+    Nothing                     -> fail $ "Expected integral, but encountered float: " <> show s
+    Just i | i >= 0 && i <= 255 -> pure $ Decimals i
+    Just _ | otherwise          -> fail $ "Decimal value must be in the range [0, 255] (inclusive)"
```

Comment:
```suggestion
validateMetadataDecimals :: MonadFail f => Int -> f Decimals
validateMetadataDecimals s =
    | i >= 0 && i <= 255 -> pure $ Decimals i
    | otherwise          -> fail $ "Decimal value must be in the range [0, 255] (inclusive)"
```